program/src/processor/sol_staker_stake_tokens.rs

```diff
@@ -120,17 +125,25 @@ pub fn process_sol_staker_stake_tokens<'a>(
     // Transfer the tokens to the vault (stakes them).
     drop(mint_data);
     drop(vault_data);
-    spl_token_2022::onchain::invoke_transfer_checked(
-        &spl_token_2022::ID,
-        ctx.accounts.source_token_account.clone(),
-        ctx.accounts.mint.clone(),
-        ctx.accounts.vault.clone(),
-        ctx.accounts.source_token_account_authority.clone(),
-        ctx.remaining_accounts,
-        amount,
-        decimals,
-        &[],
+    invoke(
+        &transfer_checked(
+            &spl_token::ID,
+            ctx.accounts.source_token_account.key,
+            ctx.accounts.mint.key,
+            ctx.accounts.vault.key,
+            ctx.accounts.source_token_account_authority.key,
+            &[],
+            amount,
+            decimals,
+        )?,
+        &[
+            ctx.accounts.source_token_account.clone(),
+            ctx.accounts.mint.clone(),
+            ctx.accounts.vault.clone(),
+            ctx.accounts.source_token_account_authority.clone(),
+        ],
     )?;
 
+    // TODO: Deposit tokens into holder rewards
     Ok(())
```

Comment:
This will be in a subsequent PR?

Reply:
Correct, this PR is already huge, I will work on the deposit, withdraw and harvest holder rewards next, which is the only missing piece in the stake program.